src/http/api/views/account.posts.view.ts

```diff
@@ -0,0 +1,667 @@
+                                'posts.in_reply_to',
+                                'reposts.account_id as reposter_id',
+                                this.db.raw(`'repost' as source`),
+                                this.db.raw(
+                                    'reposts.created_at as published_date',
+                                ),
+                            )
+                            .from('reposts')
+                            .innerJoin('posts', 'posts.id', 'reposts.post_id')
+                            .where('reposts.account_id', accountId),
+                    ])
+                    .orderBy('published_date', 'desc') //Apply sorting at the union level
+                    .as('posts_with_source'),
+            )
+            .innerJoin(
+                'accounts as author_account',
+                'author_account.id',
+                'posts_with_source.author_id',
+            )
+            .leftJoin(
+                'accounts as reposter_account',
+                'reposter_account.id',
+                'posts_with_source.reposter_id',
+            )
+            .leftJoin('likes as current_user_likes', function () {
+                this.on(
+                    'current_user_likes.post_id',
+                    'posts_with_source.id',
+                ).andOnVal(
+                    'current_user_likes.account_id',
+                    '=',
+                    defaultAccountId.toString(),
+                );
+            })
+            .leftJoin('reposts as current_user_reposts', function () {
+                this.on(
+                    'current_user_reposts.post_id',
+                    'posts_with_source.id',
+                ).andOnVal(
+                    'current_user_reposts.account_id',
+                    '=',
+                    defaultAccountId.toString(),
+                );
+            })
+            .modify((query) => {
+                if (cursor) {
+                    query.where(
+                        'posts_with_source.published_date',
+                        '<',
+                        cursor,
+                    );
+                }
+            })
+            .where('posts_with_source.deleted_at', null)
+            .where('posts_with_source.in_reply_to', null)
+            .orderBy('posts_with_source.published_date', 'desc')
+            .limit(limit + 1);
+
+        const results = await query;
+
+        const hasMore = results.length > limit;
+        const paginatedResults = results.slice(0, limit);
+        const lastResult = paginatedResults[paginatedResults.length - 1];
+
+        return {
+            results: paginatedResults.map((result: GetProfileDataResultRow) =>
+                this.mapToPostDTO(result, accountId),
+            ),
+            nextCursor: hasMore ? lastResult.published_date : null,
+        };
+    }
+
+    async getPostsByRemoteLookUp(
+        defaultAccountId: number,
+        defaultAccountApId: URL,
+        handle: string,
```

Comment:
We already have the `apId` of the remote account - I don't think we need to pass `handle` here!

Reply:
Sorry if I am being blind, but we don't have `apId` of the remote account, the `defaultAccountApId` is the appId of the current user. WE need this handle to look for the remote actor and fetch their outbox.

Reply:
Maybe I need to fix the naming `defaultAccountApId` so that it's clear

Reply:
We have the `apId` in the http handler where we do
```
const apId = await lookupAPIdByHandle(ctx, handle);
```

And we could pass that in instead of the handle

Reply:
oooh...Thanks a lot 🙏

Reply:
Done!